hal/halx86/apic/apictimer.c

```diff
@@ -63,22 +69,75 @@ VOID
 NTAPI
 HalStartProfileInterrupt(IN KPROFILE_SOURCE ProfileSource)
 {
-    UNIMPLEMENTED;
-    return;
+	LVT_REGISTER LvtEntry;
+
+	/* Only handle ProfileTime */
+	if (ProfileSource == ProfileTime)
+	{
+		/* OK, we are profiling now */
+		HalIsProfiling = TRUE;
+
+		/* Set interrupt interval */
+		ApicWrite(APIC_TICR, PCR->HalReserved[0]);
```

Comment:
If HalStartProfileInterrupt is called before/without HalSetProfileInterval, this seems like it would not have been initialized. Should ApicSetTimerInterval perhaps initialize it?

Reply:
Good point. Will do that, but put initialization in separate function